src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1916,7 +1916,10 @@ try
 
             ++num_loaded_parts;
             if (res.is_broken)
+            {
+                forcefullyRemoveBrokenOutdatedPartFromZooKeeperBeforeDetaching(res.part->name);
```

Comment:
FWIW this also fixes one minor thing - possible leftovers in ZooKeeper under `replica_path/parts`, since in case of broken parts (i.e. abnormal machine restart w/o all fsync enabled) it will remove this part only from disk (X), but if there will be a covering part (Y), then X will stays forever (and I seen such znodes in production)

Reply:
> one minor thing - possible leftovers in ZooKeeper under replica_path/parts

It's not so minor, and we should never leave leftovers in the list of parts. Because these leftovers will eventually cause "part is lost forever" errors and growth of `ReplicatedDataLoss` metric.

See also `paranoidCheckForCoveredPartsInZooKeeper`, `paranoidCheckForCoveredPartsInZooKeeperOnStart` and related issues

Reply:
I mean in cases when this "lost parts" is covered by another - non lost, it should not be a problem, right?

Reply:
Yes, nothing is actually lost, but it reports "false positive data loss", and that's a problem (it may trigger some alerts and scare some people)